packages/interactivity/src/hooks.tsx

```diff
@@ -260,18 +261,30 @@ export const directive = (
 
 // Resolve the path to some property of the store object.
 const resolve = ( path, namespace ) => {
+	if ( ! namespace ) {
+		warn(
+			`The "namespace" cannot be "{}", "null" or an emtpy string. Path: ${ path }`
+		);
+		return;
+	}
 	let resolvedStore = stores.get( namespace );
 	if ( typeof resolvedStore === 'undefined' ) {
 		resolvedStore = store( namespace, undefined, {
 			lock: universalUnlock,
 		} );
 	}
-	let current = {
+	const current = {
 		...resolvedStore,
 		context: getScope().context[ namespace ],
 	};
-	path.split( '.' ).forEach( ( p ) => ( current = current[ p ] ) );
-	return current;
+	try {
+		// TODO: Support lazy/dynamically initialized stores
+		return path.split( '.' ).reduce( ( acc, key ) => acc[ key ], current );
+	} catch ( e ) {
+		warn(
+			`The path "${ path }" could not be resolved in the "${ namespace }" store.`
```

Comment:
Loading a store after the initial hydration is legit; I don't think we should add a warning that we must remove later. It's going to cause confusion.

Reply:
Removed in https://github.com/WordPress/gutenberg/pull/61409/commits/f587e3d735e400b03bf981504d9a6c9ada8af554

Reply:
Thanks, Carlos.